Sources/Datadog/DatadogCore/Context/DatadogContextProvider.swift

```diff
@@ -116,6 +134,9 @@ internal final class DatadogContextProvider {
     ///   - reader: The value reader.
     ///   - keyPath: A context's key path that supports reading from and writing to the resulting value.
     func assign<Reader>(reader: Reader, to keyPath: WritableKeyPath<DatadogContext, Reader.Value>) where Reader: ContextValueReader {
-        self.reader.append(reader: reader, receiver: keyPath)
+        write {
+            $0[keyPath: keyPath] = reader.initialValue
```

Comment:
The reader `initialValue` is actually useless, it will always be override by the read operation. I will remove it.